pkg/reconciler/kubernetes/tektonconfig/extension.go

```diff
@@ -45,9 +46,40 @@ func (oe kubernetesExtension) PreReconcile(context.Context, v1alpha1.TektonCompo
 }
 func (oe kubernetesExtension) PostReconcile(ctx context.Context, comp v1alpha1.TektonComponent) error {
 	configInstance := comp.(*v1alpha1.TektonConfig)
-	if configInstance.Spec.Profile == common.ProfileAll {
+
+	switch configInstance.Spec.Profile {
+	case common.ProfileAll:
 		return extension.CreateDashboardCR(comp, oe.operatorClientSet.OperatorV1alpha1())
+
+	case common.ProfileBasic:
+		_, err := extension.GetDashboard(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+		if err != nil {
+			return err
+		}
+		return extension.TektonDashboardCRDelete(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+
+	case common.ProfileLite:
+		dashboard, err := extension.GetDashboard(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+		if dashboard != nil {
+			err := extension.TektonDashboardCRDelete(oe.operatorClientSet.OperatorV1alpha1().TektonDashboards(), common.DashboardResourceName)
+			if err != nil {
+				return err
+			}
+		}
+		if err != nil {
```

Comment:
If dashboard exist then above err https://github.com/tektoncd/operator/pull/394/files#diff-a100309e66fd0aa1063e2e110fd0d6d97eaa6516bf10a9f61285828a7a0697c1R62 can be nil in that case this function will never be called right :thinking: or am i missing something

Reply:
I have handled this case in the reconciler now